### PR TITLE
Update to Julia 0.7

### PR DESCRIPTION
**Pull request by mpastell**
I've started updating the package to 0.7 based on #15 . I've removed most of the deprecation warnings, but there are still few tests that don't pass.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -114,6 +109,8 @@ end
 # Testsets.
 #
 
+tokentest
```
Comment (mortenpi) [inline test/runtests.jl]: Is this supposed to be a comment?

Comment (mpastell) [reply test/runtests.jl]: Sorry that was left from testing, I'll remove it

**Comment by mpastell** [comment]
I believe this should be ready to merge. It would be nice to get a tagged version soon as I need this upgrade Weave.jl .

File: src/lexers/julia.jl [hunk 2]
```diff
@@ -146,14 +146,14 @@ julia_is_triple_string_macro(ctx::Context) = julia_is_string_macro(ctx, 3)
             ],
             :strings => [
                 (r"\"", STRING, :__pop__),
-                (r"\\([\\\"\'\$nrbtfav]|(x|u|U)[a-fA-F0-9]+|\d+)", STRING_ESCAPE),
```
Comment (mpastell) [inline src/lexers/julia.jl]: I found a bug in Weave examples where this is needed: `"\\\\"` rendered incorrectly. I couldn't find a reason why this breaks in 0.7 in the release notes though.

**Comment by mortenpi** [comment]
I'll move the bugfix commits to a separate PR, since this one has quite a lot going on already. I also fixed the version bound (`0.7-beta2` should be more appropriate). But I think this is good to go, so I'll merge one the tests pass again. And I can tag soon after (once the bugfix is in as well).

**Comment by mpastell** [comment]
Thanks!

**Pull request merged**